test/test_array_list.cpp

```diff
@@ -374,6 +374,31 @@ TEST_F(ArrayListPreInitTest, remove_preserves_data_around_it) {
   ret = rcutils_array_list_get(&list, 2, &ret_data);
   EXPECT_EQ(RCUTILS_RET_OK, ret) << rcutils_get_error_string().str;
   EXPECT_EQ((uint32_t)6, ret_data) << rcutils_get_error_string().str;
+
+  ret = rcutils_array_list_remove(&list, 2);
+  EXPECT_EQ(RCUTILS_RET_OK, ret) << rcutils_get_error_string().str;
+
+  ret = rcutils_array_list_get_size(&list, &size);
+  EXPECT_EQ(RCUTILS_RET_OK, ret) << rcutils_get_error_string().str;
+  EXPECT_EQ(size, (size_t)2);
+
```

Comment:
Can we also do `rcutils_array_list_get` for index 0 and 1 here, to ensure that they are 0 and 2, respectively?

Reply:
Yes. Updated and rebased.

<details><summary> log for `valgrind ./test_array_list` </summary>
<p>

```shell
[----------] Global test environment tear-down
[==========] 35 tests from 2 test suites ran. (115 ms total)
[  PASSED  ] 35 tests.
==49168== 
==49168== HEAP SUMMARY:
==49168==     in use at exit: 0 bytes in 0 blocks
==49168==   total heap usage: 768 allocs, 768 frees, 165,188 bytes allocated
==49168== 
==49168== All heap blocks were freed -- no leaks are possible
==49168== 
==49168== For lists of detected and suppressed errors, rerun with: -s
==49168== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)


```

</p>
</details>